### ISSUE TITLE
Error parsing protocol message

### ISSUE THREAD

**Issue by alvarogomezuria** [GoogleCloudPlatform/nodejs-getting-started#92]
Hi,
I'd like to report a bug. It is very easy to reproduce.

**Relevant info:**
* nodejs -v = v6.9.2 
* npm -v = 3.10.9 

**Steps:**
 1.- $ git clone https://github.com/GoogleCloudPlatform/nodejs-getting-started.git
 2.- $ cd nodejs-getting-started/
 3.- $ npm install
 4.- $ cd 2-structured-data/
 5.- vi config.js
 6.- Set the GCLOUD_PROJECT variable
 7.- npm test

**Output:**

```
nodejs-getting-started@1.0.0 test ~/nodejs-getting-started/2-structured-data
mocha test/index.js -t 30000

  2-structured-data/
    ✓ should install dependencies (3543ms)
    app.js
      ✓ should run (4921ms)
      ✓ should redirect / to /books (134ms)
      ✓ should check config (56ms)
    books/
      api.js
        ✓ should create a book (1401ms)
        ✓ should list books (1362ms)
        ✓ should delete a book (225ms)
      crud.js
        /books
          ✓ should show a list of books (2867ms)
{ Error: Error parsing protocol message
    at Function.GrpcService.decorateError_ (/home/gomezalvaro/nodejs-getting-started/2-structured-data/node_modules/@google-cloud/common/src/grpc-service.js:488:34)
    at Object.callback (/home/gomezalvaro/nodejs-getting-started/2-structured-data/node_modules/@google-cloud/common/src/grpc-service.js:247:35)
    at /home/gomezalvaro/nodejs-getting-started/2-structured-data/node_modules/grpc/src/node/src/client.js:420:14
  code: 400,
  metadata: Metadata { _internal_repr: {} },
  message: 'Error parsing protocol message',
  response: 'Error parsing protocol message' }
          ✓ should handle error (163ms)
        /books/add
          ✓ should post to add book form (295ms)
          ✓ should show add book form (97ms)
        /books/:book/edit & /books/:book
          ✓ should update a book (282ms)
RT <!DOCTYPE html><html lang="en"><head><title>Bookshelf - Node.js on Google Cloud Platform</title><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1"><link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.3.2/css/bootstrap.min.css"></head><body><div class="navbar navbar-default"><div class="container"><div class="navbar-header"><div class="navbar-brand">Bookshelf</div></div><ul class="nav navbar-nav"><li><a href="/books">Books</a></li></ul></div></div><div class="container"><h3>Edit book</h3><form method="POST"><div class="form-group"><label for="title">Title</label><input type="text" name="title" id="title" value="my other book" class="form-control"></div><div class="form-group"><label for="author">Author</label><input type="text" name="author" id="author" class="form-control"></div><div class="form-group"><label for="publishedDate">Date Published</label><input type="text" name="publishedDate" id="publishedDate" class="form-control"></div><div class="form-group"><label for="description">Description</label><input type="text" name="description" id="description" class="form-control"></div><div class="form-group"></div><button type="submit" class="btn btn-success">Save</button></form></div></body></html>
expected <input type="text" name="title" id="title" value="my other book" class="form-control">
          ✓ should show edit book form (278ms)
          ✓ should show a book (210ms)
          ✓ should delete a book (164ms)


  15 passing (17s)_
```

**In conclusion, error message is**

```
_{ Error: Error parsing protocol message
    at Function.GrpcService.decorateError_ (/home/gomezalvaro/nodejs-getting-started/2-structured-data/node_modules/@google-cloud/common/src/grpc-service.js:488:34)
    at Object.callback (/home/gomezalvaro/nodejs-getting-started/2-structured-data/node_modules/@google-cloud/common/src/grpc-service.js:247:35)
    at /home/gomezalvaro/nodejs-getting-started/2-structured-data/node_modules/grpc/src/node/src/client.js:420:14
  code: 400,
  metadata: Metadata { _internal_repr: {} },
  message: 'Error parsing protocol message',
  response: 'Error parsing protocol message' }_
```

Let me know if you need anything else.
Best,
Alvaro

**Comment by alvarogomezuria** [comment]
Can this bug be assigned to anyone, please?
Thanks

**Comment by pcostell** [comment]
This appears to be an issue with the test framework. Here is a more detailed log output:

``
Router req.query: {"pageToken":"badrequest"}
Datastore Request: {"query":{"distinctOn":[],"kind":[{"name":"Book"}],"order":[{"property":{"name":"publishedDate"},"direction":"ASCENDING"}],"projection":[],"limit":{"value":10},"startCursor":"badrequest"},"projectId":"pcostello-cloud"}
Datastore Error: {"code":3,"metadata":{"_internal_repr":{}}}
```

You can see the `request.query.pageToken` is getting set to `"badrequest"`. gRPC will try to decode this as bytes which will fail.

**Comment by ace-n** [comment]
The point of this test seems to be that client-side code can handle a server-side error response properly. The error you've reported occurs on the server-side and seems to be deliberately invoked (by the client) in order to check that the client-side reacts correctly.

TL;DR: that "error" is supposed to happen.

**Comment by ace-n** [comment]
(Closing this issue, @alvarogomezuria feel free to re-open if my response is missing anything. Thanks!)
